src/averageAges.js

```diff
@@ -34,7 +44,18 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let mothers;
+
+  if (withChildren) {
+    mothers = people.map(x => x.mother);
+    mothers = people.filter(x => mothers.includes(x.name));
```

Comment:
don't use one variable for everything, if you need variable mothersNames just create it